app/lib/actions/katello/environment/destroy.rb

```diff
@@ -42,7 +42,11 @@ def humanized_input
 
         def finalize
           environment = ::Katello::KTEnvironment.find(input['kt_environment']['id'])
+
+          # Don't audit anything during this action as it can sometimes try to save audits to the taxonomy that was deleted.
+          Audited.auditing_enabled = false
           environment.destroy!
+          Audited.auditing_enabled = true
```

Comment:
I'm not sure we can use this method of disabling auditing since this is being run in a multi-threaded environment and this disables it globally and doesn't seem to do so in a threadsafe way.

What about something like KTEnvironment.without_auditing{  environment.destroy!  } ?  and same for below.  I'm not actually sure if this will disable audting for just the KTEnvironment class or everything that would be affected in that block, but its worth looking inot

Reply:
ya, I wasn't sure how safe this would be, that is a good point.

I tried just specifying the audits for just the org or the environment, but it looks like the audits are still created for the dependent actions, like removing the default content view and cv env, which will still be audited. 

I'll look again, I can try nesting together a bunch of classes or figuring out a way to dynamically use `without_auditing` for a class and all its dependents. Maybe overriding the destroy action for the org and environment class would be a better place to do it too.